datasets/doc/source/ref-federated_dataset.rst

```diff
@@ -0,0 +1,34 @@
+﻿federated\_dataset
```

Comment:
We probably don't need this module documents. We expect users to use `from flwr_datasets import FederatedDataset`, and never use `flwr_datasets.federated_dataset`. The same goes for the `partitioner` package: we only want users to use whatever gets exported from that package (i.e., mentioned in `__init__.__all__`), but we don't want users to depend on the modules in that package (we might want to rename/restructure them later, the module name should be treated as an implementation detail).